### PR TITLE
Rewrite mnist example

### PR DESCRIPTION
**Pull request by alexey-khorkin**
Rewrite mnist example to use new api

### CODE REVIEW

File: openfl-tutorials/interactive_api/Tensorflow_MNIST/workspace/Mnist_Classification_FL.ipynb [hunk 1]
```diff
@@ -0,0 +1,699 @@
+{
```
Comment (aleksandr-mokrov) [inline openfl-tutorials/interactive_api/Tensorflow_MNIST/workspace/Mnist_Classification_FL.ipynb:1]: Validate this file: `Unreadable Notebook: Mnist_Classification_FL.ipynb NotJSONError('Notebook does not appear to be JSON: \'{\\n "cells": [\\n {\\n "cell_type": "m...')`

File: openfl-tutorials/interactive_api/Tensorflow_MNIST/envoy_one_folder/mnist_shard_descriptor.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
+# Copyright (C) 2020-2021 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+"""Mnist Shard Descriptor."""
+
+import logging
+
+import numpy as np
+from tensorflow import keras
```
Comment (igor-davidyuk) [inline openfl-tutorials/interactive_api/Tensorflow_MNIST/envoy_one_folder/mnist_shard_descriptor.py]: Please avoid using Deep Learning frameworks in Shard Descriptor implementation.
Almost any data preparation procedure may be implemented with numpy or any other lightweight libs.
We try to limit DL framework usage to model definition and utilising their computational backend, both done on the frontend (jupyter)

File: openfl-tutorials/interactive_api/Tensorflow_MNIST/envoy_two_folder/mnist_shard_descriptor.py [hunk 3]
```diff
@@ -0,0 +1,83 @@
+# Copyright (C) 2020-2021 Intel Corporation
```
Comment (igor-davidyuk) [inline openfl-tutorials/interactive_api/Tensorflow_MNIST/envoy_two_folder/mnist_shard_descriptor.py]: No need to repeat code. I suggest leaving only one SD implementation and several envoy configs, pointing to it

File: openfl-tutorials/interactive_api/Tensorflow_MNIST/workspace/Mnist_Classification_FL.ipynb [hunk 4]
```diff
@@ -0,0 +1,699 @@
...
+    "# Federation can also determine local fqdn automatically\n",
+    "federation = Federation(client_id=client_id, director_node_fqdn=director_node_fqdn, director_port='50051', tls=False)\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "id": "47dcfab3",
+   "metadata": {},
+   "outputs": [
```
Comment (aleksandr-mokrov) [inline openfl-tutorials/interactive_api/Tensorflow_MNIST/workspace/Mnist_Classification_FL.ipynb]: Clean all outputs

File: openfl-tutorials/interactive_api/Tensorflow_MNIST/envoy_two_folder/mnist_shard_descriptor.py (hunk 3 above)
Comment (alexey-khorkin) [reply openfl-tutorials/interactive_api/Tensorflow_MNIST/envoy_two_folder/mnist_shard_descriptor.py]: I created two folders (with the same shard code) specifically for easy use, to use the two collaborators, we just need to copy the folder to another node.

File: openfl-tutorials/interactive_api/Tensorflow_MNIST/workspace/Mnist_Classification_FL.ipynb [hunk 5]
```diff
@@ -0,0 +1,407 @@
...
+   "outputs": [],
+   "source": [
+    "# The following command zips the workspace and python requirements to be transfered to collaborator nodes\n",
+    "fl_experiment.start(model_provider=MI, \n",
+    "                   task_keeper=TI,\n",
+    "                   data_loader=fed_dataset,\n",
+    "                   rounds_to_train=5,\n",
+    "                   opt_treatment='CONTINUE_GLOBAL')"
+   ]
+  },
```
Comment (itrushkin) [inline openfl-tutorials/interactive_api/Tensorflow_MNIST/workspace/Mnist_Classification_FL.ipynb:376]: Might be useful to see metrics with `fl_experiment.stream_metrics()`.

File: openfl-tutorials/interactive_api/Tensorflow_MNIST/workspace/Mnist_Classification_FL.ipynb [hunk 6]
```diff
@@ -0,0 +1,407 @@
...
+   "outputs": [],
+   "source": []
+  }
+ ],
+ "metadata": {
+  "kernelspec": {
+   "display_name": "openfl",
+   "language": "python",
+   "name": "openfl"
+  },
```
Comment (itrushkin) [inline openfl-tutorials/interactive_api/Tensorflow_MNIST/workspace/Mnist_Classification_FL.ipynb]: Please clean Python kernel with 
```
jupyter nbconvert --ClearOutputPreprocessor.enabled=True --ClearMetadataPreprocessor.enabled=True --inplace /your/path/to/the/notebook.ipynb
```

**Review by itrushkin — CHANGES_REQUESTED** [review]

**Review by itrushkin — APPROVED** [review]

**Review by aleksandr-mokrov — APPROVED** [review]

**Pull request merged**